### PR TITLE
Print cURL command in PA

### PR DESCRIPTION
**Pull request by lkomali**
This PR modifies openai_client to print cURL command in extra verbose mode in PA for debugging purposes.

image

### CODE REVIEW

**Comment by dyastremsky** [comment]
Do we not want the HTTP client to also print the cURL command?

File: src/client_backend/openai/http_client.h [hunk 1]
```diff
@@ -142,7 +147,6 @@ class HttpClient {
   // Note that this function does not block
   void Send(CURL* handle, std::unique_ptr<HttpRequest>&& request);
 
- protected:
```
Comment (dyastremsky) [inline src/client_backend/openai/http_client.h:145]: Is there a reason we no longer want this information to be protected?

**Review by dyastremsky — COMMENTED** [review]
Great work on this PR! Nice job creating the new test as well.

Left a couple of questions.

File: src/client_backend/openai/http_client.h (hunk 1 above)
Comment (nicolasnoble) [reply src/client_backend/openai/http_client.h:145]: I'm guessing it's to access `total_input_byte_size_` from `ChatCompletionClient::PrintCurlCommand`. It'd probably be better to add a public accessor for it instead.

**Comment by lkomali** [comment]
@dyastremsky That's what @nv-hwoo raised in the slack thread, as to why we want to print the command twice. 
So I retained the Print function in the openai_client.

File: src/client_backend/openai/http_client.h (hunk 1 above)
Comment (lkomali) [reply src/client_backend/openai/http_client.h:145]: Aren't they in protected ? We have defined at line 140 and all the others that follow are protected right? 
I removed thinking it's a redundant declaration.

Comment (nicolasnoble) [reply src/client_backend/openai/http_client.h:145]: Oh, derp, totally missed the fact this `protected:` here is redundant. Proceed ^_^

Comment (lkomali) [reply src/client_backend/openai/http_client.h:145]: Is it why I removed the line 145? I think it's redundant and that's why I removed. we have one protected in line 140 and everything that follows till line 171 is protected?

Comment (nicolasnoble) [reply src/client_backend/openai/http_client.h:145]: Yep, our bad, we just haven't been paying attention.

**Comment by nv-hwoo** [comment]
@lkomali could you add the output of the curl command (e.g. screenshot) to the PR? Would like to see the final look.

**Comment by lkomali** [comment]
@nv-hwoo Attached the screenshot in the description.

**Comment by dyastremsky** [comment]
> @dyastremsky That's what @nv-hwoo raised in the slack thread, as to why we want to print the command twice. So I retained the Print function in the openai_client.

OpenAI isn't the only client we use though. We also use the HTTP client. Wouldn't users want to see the cURL command for the HTTP client too? A solution that allows the cURL command to be printed for all of the relevant clients would be best.

**Review by dyastremsky — APPROVED** [review]

**Pull request merged**